src/javascript/_common/os_detect.js

```diff
@@ -0,0 +1,23 @@
+const OSDetect = () => {
+    const Systems = {
+        macos  : ['Mac68K', 'MacIntel', 'MacPPC'],
+        linux  : ['HP-UX', 'Linux i686', 'Linux armv7l', ''],
+        ios    : ['iPhone', 'iPod', 'iPad'],
+        android: ['Android'],
+        windows: ['Win16', 'Win32', 'WinCE'],
+    };
```

Comment:
might be better to declare this outside, then we don't have to re-declare it every time the function is called